### DISCUSSION TITLE
Make a way for custom calendar integrations to assign the color of calendar entities

### DISCUSSION THREAD

**Discussion by moofur** [home-assistant/frontend#18465] [Other feature requests]
I am someone who uses the O365-HomeAssistant custom integration (https://github.com/RogerSelwyn/O365-HomeAssistant/). I specifically use it to pull calendar data from my and other shared/family calendars into Home Assistant for display on dashboards using the standard "Calendar" card.

The developer over there just released an update that pulls the color of the calendar, the color that the user has assigned the calendar in Office 365 online. The integration pulls a hex value for the assigned color. I understand that HA won't just conform Calendar functionality to that custom integration, but is there a way to build a bridge that custom integrations like this one can use to plug in the desired pre-assigned color for an individual calendar entity?

I'm including a screenshot of the Attributes data (scrubbed) from one of my calendar entities.
![image](https://github.com/home-assistant/frontend/assets/22972834/e66ffda9-cdb2-4c3d-94de-da8a4c98356f)

**Comment by moofur** [comment]
This may be something that could happen after this: https://github.com/home-assistant/architecture/discussions/883
1. Implement the ability to assign a custom color to each individual calendar entity
2. Implement a way for custom integrations to be able to provide the desired color for each calendar entity

**Comment by frenck** [comment]
This is not a frontend feature request, this requires backend changes and architectural discussions. Closing the request on this end for that reason.

../Frenck
